### PR TITLE
Add the Nod mobile stealth generator

### PR DESCRIPTION
**Pull request by Mailaender**
Based on
* https://modenc.renegadeprojects.com/Implementing_Stealth_Generators
* https://cnc.fandom.com/wiki/Mobile_stealth_generator
* https://github.com/CnCNet/cncnet-ts-client-package/blob/master/INI/rules.ini

### CODE REVIEW

**Comment by JovialFeline** [comment]
I'm less familiar with TS but I gave this a spin and it seems okay. I only noted that it lacks a range circle when deployed.

File: mods/ts/rules/nod-vehicles.yaml [hunk 1]
```diff
@@ -547,3 +547,84 @@ STNK:
...
 		Weapon: UnitExplode
+
+SGEN:
+	Inherits: ^Tank
+	Inherits@VOXELS: ^VoxelActor
+	Inherits@selection: ^SelectableSupportUnit
+	-AppearsOnRadar:
+	Buildable:
+		Queue: Vehicle
+		BuildPaletteOrder: 90
```
Comment (PunkPun) [inline mods/ts/rules/nod-vehicles.yaml]: it should be 100+. This is only buildable after tech centre so units with lower prerequisites such as Nod Artillery (100) should come first

File: mods/ts/rules/nod-vehicles.yaml [hunk 2]
```diff
@@ -547,3 +547,84 @@ STNK:
...
+		Facing: 384
+		AllowedTerrainTypes: Clear, Road, DirtRoad, Rough
+		DeploySounds: place2.aud
+		UndeploySounds: clicky1.aud
+		Voice: Move
+	EntersTunnels:
+		RequireForceMoveCondition: !undeployed
+	Repairable:
+		RequireForceMoveCondition: !undeployed
+	Passenger:
```
Comment (PunkPun) [inline mods/ts/rules/nod-vehicles.yaml:613]: a bunch of these traits should have RequiresCondition: !empdisable

I managed to undeploy an EMP'ed generator by force moving

File: mods/ts/rules/nod-vehicles.yaml [hunk 3]
```diff
@@ -547,3 +547,84 @@ STNK:
...
+	Voiced:
+		VoiceSet: Spy
+	Mobile:
+		Speed: 85
+		TurnSpeed: 20
+		ImmovableCondition: !undeployed
+		RequireForceMoveCondition: !undeployed
+	RevealsShroud:
+		RequiresCondition: !inside-tunnel && undeployed
+		Range: 9c0
```
Comment (PunkPun) [inline mods/ts/rules/nod-vehicles.yaml]: shouldn't be 5 as in the wiki?

File: mods/ts/rules/nod-vehicles.yaml [hunk 4]
```diff
@@ -547,3 +547,88 @@ STNK:
...
+		UndeployedCondition: undeployed
+		UndeployOnMove: true
+		UndeployOnPickup: true
+		Facing: 384
+		AllowedTerrainTypes: Clear, Road, DirtRoad, Rough
+		DeploySounds: place2.aud
+		UndeploySounds: clicky1.aud
+		Voice: Move
+	EntersTunnels:
+		RequireForceMoveCondition: !undeployed && !empdisable
```
Comment (PunkPun) [inline mods/ts/rules/nod-vehicles.yaml]:
```suggestion
		RequireForceMoveCondition: !undeployed
```

That's not how it works. The `empdisable` should on `RequiresCondition` or `PauseOnCondition`, though it seems that none of these traits are conditional. Repairable will become conditional with the docking refactor, but others won't

Comment (Mailaender) [reply mods/ts/rules/nod-vehicles.yaml:608]: You explicitly asked for this in the comment before...

File: mods/ts/rules/nod-vehicles.yaml (hunk 2 above)
Comment (Mailaender) [reply mods/ts/rules/nod-vehicles.yaml:613]: What bunch of traits. Be more specific.

File: mods/ts/rules/nod-vehicles.yaml (hunk 4 above)
Comment (Mailaender) [reply mods/ts/rules/nod-vehicles.yaml]: You are now making me apply changes and revert them again.

Comment (PunkPun) [reply mods/ts/rules/nod-vehicles.yaml]: I didn't check to see wether they were conditional

**Comment by PunkPun** [comment]
grafic

is there a way we could make it centred?

**Comment by abcdefg30** [comment]
I agree, there seems to be some kind of visual "jump upwards" when deploying. Can we offset the artwork a bit?

**Comment by penev92** [comment]
I'm going to leave this here for the record so I don't have to search for it again and again:
image

**Comment by Mailaender** [comment]
Adjusted the offset `+1, -1` measured with @asaeprite based on the `.gif`.

**Comment by PunkPun** [comment]
New codestyle failed.

Also the sprite seems a pixel off and shadow seems very off. Is that how it was in vanilla? Though I suppose we will need to change the way shadows are in TS anyhow, so this is a mute point

gif

**Comment by PunkPun** [comment]
Though I suppose we will need to change the way shadows are in TS anyhow, so this is a mute point

**Comment by PunkPun** [comment]
in the wiki I found this, it seems the shadow is rendered differently for the undeployed version. I suppose the shadow might be the primary reason why the generator feels like teleporting

gifgif

**Comment by PunkPun** [comment]
I suppose the shadow might be the primary reason why the generator feels like teleporting

**Review by PunkPun — APPROVED** [review]

**Comment by PunkPun** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/15196c52ca487b052a73d405bb1cdab86355c0d2...60069dfae1ba8e2e9cb0ab1093e96aa5f00089d1)

**Pull request merged**